
<!DOCTYPE html>
<html lang="en" class="loading">
<head>
    <meta charset="UTF-8" />
    <meta http-equiv="X-UA-Compatible" content="IE=edge,chrome=1" />
    <meta name="viewport" content="width=device-width, minimum-scale=1.0, maximum-scale=1.0, user-scalable=no">
    <title>单细胞实战(4)：STAR与cellranger结果比较 - Zhz Blog</title>
    <meta name="apple-mobile-web-app-capable" content="yes" />
    <meta name="apple-mobile-web-app-status-bar-style" content="black-translucent">
    <meta name="google" content="notranslate" />
    <meta name="keywords" content="zhoushizhao,"> 
    <meta name="description" content="前言本次主要使用Seurat比较STAR与Cellranger的输出结果，只会进行简单的聚类工作。
数据读入123456789101112131415161718192021222324252627,"> 
    <meta name="author" content="John Doe"> 
    <link rel="alternative" href="atom.xml" title="Zhz Blog" type="application/atom+xml"> 
    <link rel="icon" href="/img/favicon.png"> 
    
    
    
    <meta name="twitter:card" content="summary"/>
    <meta name="twitter:title" content="单细胞实战(4)：STAR与cellranger结果比较 - Zhz Blog"/>
    <meta name="twitter:description" content="前言本次主要使用Seurat比较STAR与Cellranger的输出结果，只会进行简单的聚类工作。
数据读入123456789101112131415161718192021222324252627,"/>
    
    
    
    
    <meta property="og:site_name" content="Zhz Blog"/>
    <meta property="og:type" content="object"/>
    <meta property="og:title" content="单细胞实战(4)：STAR与cellranger结果比较 - Zhz Blog"/>
    <meta property="og:description" content="前言本次主要使用Seurat比较STAR与Cellranger的输出结果，只会进行简单的聚类工作。
数据读入123456789101112131415161718192021222324252627,"/>
    
<link rel="stylesheet" href="/css/diaspora.css">

<meta name="generator" content="Hexo 5.4.1"></head>

<body class="loading">
    <span id="config-title" style="display:none">Zhz Blog</span>
    <div id="loader"></div>
    <div id="single">
    <div id="top" style="display: block;">
    <div class="bar" style="width: 0;"></div>
    <a class="iconfont icon-home image-icon" href="javascript:;" data-url="http://example.com"></a>
    <div title="播放/暂停" class="iconfont icon-play"></div>
    <h3 class="subtitle">单细胞实战(4)：STAR与cellranger结果比较</h3>
    <div class="social">
        <div>
            <div class="share">
                <a title="获取二维码" class="iconfont icon-scan" href="javascript:;"></a>
            </div>
            <div id="qr"></div>
        </div>
    </div>
    <div class="scrollbar"></div>
</div>

    <div class="section">
        <div class="article">
    <div class='main'>
        <h1 class="title">单细胞实战(4)：STAR与cellranger结果比较</h1>
        <div class="stuff">
            <span>十月 06, 2020</span>
            
  <ul class="post-tags-list" itemprop="keywords"><li class="post-tags-list-item"><a class="post-tags-list-link" href="/tags/ScRNAseq/" rel="tag">ScRNAseq</a></li><li class="post-tags-list-item"><a class="post-tags-list-link" href="/tags/bioinformatics/" rel="tag">bioinformatics</a></li></ul>


        </div>
        <div class="content markdown">
            <h2 id="前言"><a href="#前言" class="headerlink" title="前言"></a>前言</h2><p>本次主要使用Seurat比较STAR与Cellranger的输出结果，只会进行简单的聚类工作。</p>
<h2 id="数据读入"><a href="#数据读入" class="headerlink" title="数据读入"></a>数据读入</h2><figure class="highlight plaintext"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br><span class="line">7</span><br><span class="line">8</span><br><span class="line">9</span><br><span class="line">10</span><br><span class="line">11</span><br><span class="line">12</span><br><span class="line">13</span><br><span class="line">14</span><br><span class="line">15</span><br><span class="line">16</span><br><span class="line">17</span><br><span class="line">18</span><br><span class="line">19</span><br><span class="line">20</span><br><span class="line">21</span><br><span class="line">22</span><br><span class="line">23</span><br><span class="line">24</span><br><span class="line">25</span><br><span class="line">26</span><br><span class="line">27</span><br><span class="line">28</span><br><span class="line">29</span><br><span class="line">30</span><br><span class="line">31</span><br><span class="line">32</span><br><span class="line">33</span><br></pre></td><td class="code"><pre><span class="line">&gt; library(Seurat)</span><br><span class="line">&gt; library(dplyr)</span><br><span class="line">&gt; library(magrittr)</span><br><span class="line">&gt; library(gtools)</span><br><span class="line">&gt; library(stringr)</span><br><span class="line">&gt; library(Matrix)</span><br><span class="line">&gt; setwd(&quot;D://data/ScRNAcode&quot;)</span><br><span class="line">##读入STAR数据</span><br><span class="line">&gt; matrix.dir=&quot;STAR/&quot;</span><br><span class="line">&gt; barcode.path &lt;- paste0(matrix.dir,&quot;barcodes.tsv&quot;)</span><br><span class="line">&gt; features.path &lt;- paste0(matrix.dir,&quot;features.tsv&quot;)</span><br><span class="line">&gt; matrix.path &lt;- paste0(matrix.dir, &quot;matrix.mtx&quot;)</span><br><span class="line">&gt; STARmatrix &lt;- readMM(file = matrix.path)</span><br><span class="line">&gt; feature.names = read.delim(features.path,</span><br><span class="line">+                            header = FALSE,</span><br><span class="line">+                            stringsAsFactors = FALSE)</span><br><span class="line">&gt; barcode.names = read.delim(barcode.path,</span><br><span class="line">+                            header = FALSE,</span><br><span class="line">+                            stringsAsFactors = FALSE)</span><br><span class="line">&gt; colnames(STARmatrix) = barcode.names$V1</span><br><span class="line">&gt; rownames(STARmatrix) = feature.names$V2</span><br><span class="line">&gt; STARmatrix&lt;-as.matrix(STARmatrix)</span><br><span class="line">&gt; STARmatrix[1:6,1:6]</span><br><span class="line">##创建seurat对象</span><br><span class="line">##创建STAR的Seurat对象</span><br><span class="line">&gt; STAR &lt;- CreateSeuratObject(STARmatrix,project = &quot;zsz&quot;,</span><br><span class="line">                           min.cells = 3, min.features = 200)</span><br><span class="line">##创建Cellranger的Seurat对象</span><br><span class="line">&gt; dir=&quot;cellranger/&quot;</span><br><span class="line">&gt; counts &lt;- Read10X(data.dir = dir)</span><br><span class="line">&gt; RANGER &lt;- CreateSeuratObject(counts, project = &quot;zsz&quot;,</span><br><span class="line">                             min.cells=3, min.features = 200)</span><br><span class="line"></span><br></pre></td></tr></table></figure>

<h2 id="数据比较"><a href="#数据比较" class="headerlink" title="数据比较"></a>数据比较</h2><figure class="highlight plaintext"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br><span class="line">7</span><br><span class="line">8</span><br><span class="line">9</span><br><span class="line">10</span><br><span class="line">11</span><br><span class="line">12</span><br><span class="line">13</span><br><span class="line">14</span><br><span class="line">15</span><br><span class="line">16</span><br><span class="line">17</span><br><span class="line">18</span><br><span class="line">19</span><br><span class="line">20</span><br><span class="line">21</span><br><span class="line">22</span><br><span class="line">23</span><br><span class="line">24</span><br><span class="line">25</span><br><span class="line">26</span><br><span class="line">27</span><br><span class="line">28</span><br><span class="line">29</span><br><span class="line">30</span><br><span class="line">31</span><br><span class="line">32</span><br><span class="line">33</span><br></pre></td><td class="code"><pre><span class="line">&gt; ##数据比较</span><br><span class="line">&gt; dim(STARmatrix)</span><br><span class="line">[1] 33538  2048</span><br><span class="line">&gt; dim(counts)</span><br><span class="line">[1] 33538  2112</span><br><span class="line">&gt; dim(STAR)</span><br><span class="line">[1] 13350  2046</span><br><span class="line">&gt; dim(RANGER)</span><br><span class="line">[1] 13314  2105</span><br><span class="line">&gt; fivenum(apply(STARmatrix,1,function(x) sum(x&gt;0)))</span><br><span class="line">MIR1302-2HG     IGFBPL1  AL008723.2       FXYD7      MT-CO1</span><br><span class="line">          0           0           0          29        2047</span><br><span class="line">&gt; fivenum(apply(counts,1,function(x) sum(x&gt;0)))</span><br><span class="line">MIR1302-2HG     FAM221B   LINC01638        DGKE      MT-CO1</span><br><span class="line">          0           0           0          29        2108</span><br><span class="line">&gt; pdf(&quot;box.pdf&quot;,height = 9,width = 9)</span><br><span class="line">&gt; boxplot(apply(STARmatrix,1,function(x) sum(x&gt;0) ),main = &quot;STAR&quot;,col = &quot;lightgray&quot;)</span><br><span class="line">&gt; boxplot(apply(counts,1,function(x) sum(x&gt;0) ),main = &quot;Cellranger&quot;,col = &quot;lightgray&quot;)</span><br><span class="line">&gt; dev.off()</span><br><span class="line">RStudioGD</span><br><span class="line">        2</span><br><span class="line">&gt; pdf(&quot;hist.pdf&quot;,height = 9,width = 9)</span><br><span class="line">&gt; hist(apply(STARmatrix,2,function(x) sum(x&gt;0) ),col = &quot;lightgray&quot;,</span><br><span class="line">+      breaks=20,xlim=c(0,4000),ylim=c(0,800),</span><br><span class="line">+      labels=F,main=&quot;STAR&quot;,</span><br><span class="line">+      xlab=&quot;genes&quot;,ylab=&quot;cells&quot;)</span><br><span class="line">&gt; abline(v=median(apply(STARmatrix,2,function(x) sum(x&gt;0))),col=&#x27;red&#x27;)</span><br><span class="line">&gt; hist(apply(counts,2,function(x) sum(x&gt;0) ),col = &quot;lightgray&quot;,</span><br><span class="line">+      breaks=20,xlim=c(0,4000),ylim=c(0,800),</span><br><span class="line">+      labels=F,main=&quot;Cellranger&quot;,</span><br><span class="line">+      xlab=&quot;genes&quot;,ylab=&quot;cells&quot;)</span><br><span class="line">&gt; abline(v=median(apply(counts,2,function(x) sum(x&gt;0))),col=&#x27;red&#x27;)</span><br><span class="line">&gt; dev.off()              </span><br></pre></td></tr></table></figure>

<p><img src="/img/posts/2020.10.8/image-20201019104953848.png" alt="image-20201019104953848"></p>
<p><img src="/img/posts/2020.10.8/image-20201019105730957.png" alt="image-20201019105730957"></p>
<p>根据箱线图，直方图和矩阵的基本信息，可以看到STAR与cellranger的结果差距很小</p>
<h2 id="质量控制与聚类比较"><a href="#质量控制与聚类比较" class="headerlink" title="质量控制与聚类比较"></a>质量控制与聚类比较</h2><figure class="highlight plaintext"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br><span class="line">7</span><br><span class="line">8</span><br><span class="line">9</span><br><span class="line">10</span><br><span class="line">11</span><br><span class="line">12</span><br></pre></td><td class="code"><pre><span class="line">&gt; pdf(&quot;qc.pdf&quot;,height = 9,width = 9)</span><br><span class="line">&gt; VlnPlot(STAR,</span><br><span class="line">+         features = c(&quot;nFeature_RNA&quot;, &quot;nCount_RNA&quot;),</span><br><span class="line">+         pt.size = 0.1,</span><br><span class="line">+         ncol = 2)</span><br><span class="line">&gt; VlnPlot(RANGER,</span><br><span class="line">+         features = c(&quot;nFeature_RNA&quot;, &quot;nCount_RNA&quot;),</span><br><span class="line">+         pt.size = 0.1,</span><br><span class="line">+         ncol = 2)</span><br><span class="line">&gt; dev.off()</span><br><span class="line">RStudioGD</span><br><span class="line">        2</span><br></pre></td></tr></table></figure>

<p><img src="/img/posts/2020.10.8/image-20201019140112013.png" alt="image-20201019140112013"></p>
<figure class="highlight plaintext"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br><span class="line">7</span><br><span class="line">8</span><br><span class="line">9</span><br><span class="line">10</span><br><span class="line">11</span><br><span class="line">12</span><br><span class="line">13</span><br><span class="line">14</span><br><span class="line">15</span><br><span class="line">16</span><br><span class="line">17</span><br><span class="line">18</span><br><span class="line">19</span><br><span class="line">20</span><br><span class="line">21</span><br><span class="line">22</span><br><span class="line">23</span><br><span class="line">24</span><br><span class="line">25</span><br><span class="line">26</span><br><span class="line">27</span><br><span class="line">28</span><br><span class="line">29</span><br><span class="line">30</span><br><span class="line">31</span><br><span class="line">32</span><br><span class="line">33</span><br><span class="line">34</span><br><span class="line">35</span><br><span class="line">36</span><br><span class="line">37</span><br><span class="line">38</span><br><span class="line">39</span><br><span class="line">40</span><br><span class="line">41</span><br><span class="line">42</span><br><span class="line">43</span><br><span class="line">44</span><br><span class="line">45</span><br><span class="line">46</span><br><span class="line">47</span><br><span class="line">48</span><br><span class="line">49</span><br><span class="line">50</span><br><span class="line">51</span><br><span class="line">52</span><br><span class="line">53</span><br><span class="line">54</span><br><span class="line">55</span><br><span class="line">56</span><br></pre></td><td class="code"><pre><span class="line">&gt; STAR&lt;-subset(STAR,subset=nFeature_RNA&gt;500 &amp; nFeature_RNA&lt;2000)</span><br><span class="line">&gt; STAR &lt;- NormalizeData(STAR, normalization.method = &quot;LogNormalize&quot;, scale.factor = 10000)</span><br><span class="line">Performing log-normalization</span><br><span class="line">0%   10   20   30   40   50   60   70   80   90   100%</span><br><span class="line">[----|----|----|----|----|----|----|----|----|----|</span><br><span class="line">**************************************************|</span><br><span class="line">&gt; table(STAR@meta.data$orig.ident)</span><br><span class="line"></span><br><span class="line"> zsz</span><br><span class="line">1896</span><br><span class="line">&gt; STAR &lt;- FindVariableFeatures(STAR, selection.method = &quot;vst&quot;, nfeatures = 2000)</span><br><span class="line">Calculating gene variances</span><br><span class="line">0%   10   20   30   40   50   60   70   80   90   100%</span><br><span class="line">[----|----|----|----|----|----|----|----|----|----|</span><br><span class="line">**************************************************|</span><br><span class="line">Calculating feature variances of standardized and clipped values</span><br><span class="line">0%   10   20   30   40   50   60   70   80   90   100%</span><br><span class="line">[----|----|----|----|----|----|----|----|----|----|</span><br><span class="line">**************************************************|</span><br><span class="line">&gt; top10 &lt;- head(VariableFeatures(STAR), 10)</span><br><span class="line">&gt; top10</span><br><span class="line"> [1] &quot;S100A9&quot;  &quot;S100A8&quot;  &quot;IGLC2&quot;   &quot;IGKC&quot;    &quot;LYZ&quot;     &quot;IGLC3&quot;   &quot;CCL3&quot;    &quot;NFKBIA&quot;  &quot;PTGDS&quot;   &quot;S100A12&quot;</span><br><span class="line"></span><br><span class="line"> &gt; RANGER&lt;-subset(RANGER,subset=nFeature_RNA&gt;500 &amp; nFeature_RNA&lt;2000)</span><br><span class="line">&gt; RANGER&lt;- NormalizeData(RANGER, normalization.method = &quot;LogNormalize&quot;, scale.factor = 10000)</span><br><span class="line">Performing log-normalization</span><br><span class="line">0%   10   20   30   40   50   60   70   80   90   100%</span><br><span class="line">[----|----|----|----|----|----|----|----|----|----|</span><br><span class="line">**************************************************|</span><br><span class="line">&gt; table(RANGER@meta.data$orig.ident)</span><br><span class="line"></span><br><span class="line"> zsz</span><br><span class="line">1895</span><br><span class="line">&gt; RANGER &lt;- FindVariableFeatures(RANGER, selection.method = &quot;vst&quot;, nfeatures = 2000)</span><br><span class="line">Calculating gene variances</span><br><span class="line">0%   10   20   30   40   50   60   70   80   90   100%</span><br><span class="line">[----|----|----|----|----|----|----|----|----|----|</span><br><span class="line">**************************************************|</span><br><span class="line">Calculating feature variances of standardized and clipped values</span><br><span class="line">0%   10   20   30   40   50   60   70   80   90   100%</span><br><span class="line">[----|----|----|----|----|----|----|----|----|----|</span><br><span class="line">**************************************************|</span><br><span class="line">&gt; top10 &lt;- head(VariableFeatures(RANGER), 10)</span><br><span class="line">&gt; top10</span><br><span class="line"> [1] &quot;S100A9&quot;  &quot;S100A8&quot;  &quot;IGLC2&quot;   &quot;IGKC&quot;    &quot;LYZ&quot;     &quot;CCL3&quot;    &quot;NFKBIA&quot;  &quot;IGLC3&quot;   &quot;PTGDS&quot;   &quot;S100A12&quot;</span><br><span class="line"></span><br><span class="line">&gt; scale.genes &lt;-  rownames(STAR)</span><br><span class="line">&gt; STAR &lt;- ScaleData(STAR, features = scale.genes)</span><br><span class="line">&gt; STAR &lt;- RunPCA(STAR, features = VariableFeatures(STAR))</span><br><span class="line">&gt; plot1 &lt;- ElbowPlot(STAR, ndims=30, reduction=&quot;pca&quot;)</span><br><span class="line">&gt; scale.genes &lt;-  rownames(RANGER)</span><br><span class="line">&gt; RANGER &lt;- ScaleData(RANGER, features = scale.genes)</span><br><span class="line">&gt; RANGER &lt;- RunPCA(RANGER, features = VariableFeatures(RANGER))</span><br><span class="line">&gt; plot2 &lt;- ElbowPlot(RANGER, ndims=30, reduction=&quot;pca&quot;)</span><br><span class="line">&gt; plotc &lt;- plot1+plot2</span><br><span class="line">&gt; ggsave(&quot;pca.pdf&quot;, plot = plotc, width = 8, height = 4)</span><br></pre></td></tr></table></figure>

<p><img src="/img/posts/2020.10.8/image-20201019142612848.png" alt="image-20201019142612848"></p>
<figure class="highlight plaintext"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br><span class="line">7</span><br><span class="line">8</span><br><span class="line">9</span><br><span class="line">10</span><br><span class="line">11</span><br><span class="line">12</span><br><span class="line">13</span><br><span class="line">14</span><br><span class="line">15</span><br><span class="line">16</span><br><span class="line">17</span><br><span class="line">18</span><br><span class="line">19</span><br><span class="line">20</span><br><span class="line">21</span><br><span class="line">22</span><br><span class="line">23</span><br></pre></td><td class="code"><pre><span class="line">&gt; STAR &lt;- FindNeighbors(STAR, dims = 1:10)</span><br><span class="line">&gt; STAR &lt;- FindClusters(STAR, resolution = 0.8)</span><br><span class="line">&gt; table(STAR@meta.data$seurat_clusters)</span><br><span class="line">  0   1   2   3   4   5   6   7   8   9</span><br><span class="line">368 310 290 217 184 129 119 112  92  75</span><br><span class="line">&gt; metadata &lt;- STAR@meta.data</span><br><span class="line">&gt; cell_cluster &lt;-data.frame(cell_ID=rownames(metadata),</span><br><span class="line">cluster_ID=metadata$seurat_clusters)</span><br><span class="line">&gt; STAR &lt;- RunUMAP(STAR, dims = 1:20)</span><br><span class="line">&gt; embed_tsne &lt;- Embeddings(STAR, &#x27;umap&#x27;)</span><br><span class="line">&gt; plot1 = DimPlot(STAR, reduction = &quot;umap&quot; ,label = &quot;T&quot;, pt.size = 1,label.size = 4)</span><br><span class="line">&gt; RANGER &lt;- FindNeighbors(RANGER, dims = 1:10)</span><br><span class="line">&gt; RANGER &lt;- FindClusters(RANGER, resolution = 0.8)</span><br><span class="line">&gt; table(RANGER@meta.data$seurat_clusters)</span><br><span class="line">  0   1   2   3   4   5   6   7   8   9</span><br><span class="line">375 300 290 212 195 128 122 108  91  74</span><br><span class="line">&gt; metadata &lt;- RANGER@meta.data</span><br><span class="line">&gt; cell_cluster &lt;- data.frame(cell_ID=rownames(metadata), cluster_ID=metadata$seurat_clusters)</span><br><span class="line">&gt; RANGER &lt;- RunUMAP(RANGER,n.neighbors = 30,dims = 1:20)</span><br><span class="line">&gt; embed_umap &lt;- Embeddings(RANGER, &#x27;umap&#x27;)</span><br><span class="line">&gt; plot2 = DimPlot(RANGER, reduction = &quot;umap&quot; ,label = &quot;T&quot;, pt.size = 1,label.size = 4)</span><br><span class="line">&gt; plotc &lt;- plot1+plot2</span><br><span class="line">&gt; ggsave(&quot;umap.pdf&quot;, plot = plotc, width = 8, height = 4)</span><br></pre></td></tr></table></figure>

<p><img src="/img/posts/2020.10.8/image-20201019143534097.png" alt="image-20201019143534097"></p>
<p><img src="/img/posts/2020.10.8/image-20201019143610792.png" alt="image-20201019143610792"></p>
<p>可以看到两种分析方法umap与tsne聚类效果不太相同，但是基本的聚类与分群是一致的</p>
<h2 id="结语"><a href="#结语" class="headerlink" title="结语"></a>结语</h2><p>就结果而言，两种分析方法的结果不完全相同，但也基本一致，本次笔记中用到的数据和代码已上传<a target="_blank" rel="noopener" href="https://github.com/Bioinformatics-rookie/ScRNAseq_code">github</a>，在ScRNAseq_code/compare文件夹下，大家需要的可以下载试试</p>

            <!--[if lt IE 9]><script>document.createElement('audio');</script><![endif]-->
            <audio id="audio" loop="1" preload="auto" controls="controls" data-autoplay="false">
                <source type="audio/mpeg" src="">
            </audio>
            
                <ul id="audio-list" style="display:none">
                    
                        
                            <li title="0" data-url="http://link.hhtjim.com/163/425570952.mp3"></li>
                        
                    
                        
                            <li title="1" data-url="http://link.hhtjim.com/163/425570952.mp3"></li>
                        
                    
                </ul>
            
        </div>
        
    <div id="gitalk-container" class="comment link"
		data-enable="false"
        data-ae="false"
        data-ci=""
        data-cs=""
        data-r=""
        data-o=""
        data-a=""
        data-d="false"
    >查看评论</div>


    </div>
    
</div>


    </div>
</div>
</body>


<script src="//lib.baomitu.com/jquery/1.8.3/jquery.min.js"></script>
<script src="/js/plugin.js"></script>
<script src="/js/typed.js"></script>
<script src="/js/diaspora.js"></script>


<link rel="stylesheet" href="/photoswipe/photoswipe.css">
<link rel="stylesheet" href="/photoswipe/default-skin/default-skin.css">


<script src="/photoswipe/photoswipe.min.js"></script>
<script src="/photoswipe/photoswipe-ui-default.min.js"></script>


<!-- Root element of PhotoSwipe. Must have class pswp. -->
<div class="pswp" tabindex="-1" role="dialog" aria-hidden="true">
    <!-- Background of PhotoSwipe. 
         It's a separate element as animating opacity is faster than rgba(). -->
    <div class="pswp__bg"></div>
    <!-- Slides wrapper with overflow:hidden. -->
    <div class="pswp__scroll-wrap">
        <!-- Container that holds slides. 
            PhotoSwipe keeps only 3 of them in the DOM to save memory.
            Don't modify these 3 pswp__item elements, data is added later on. -->
        <div class="pswp__container">
            <div class="pswp__item"></div>
            <div class="pswp__item"></div>
            <div class="pswp__item"></div>
        </div>
        <!-- Default (PhotoSwipeUI_Default) interface on top of sliding area. Can be changed. -->
        <div class="pswp__ui pswp__ui--hidden">
            <div class="pswp__top-bar">
                <!--  Controls are self-explanatory. Order can be changed. -->
                <div class="pswp__counter"></div>
                <button class="pswp__button pswp__button--close" title="Close (Esc)"></button>
                <button class="pswp__button pswp__button--share" title="Share"></button>
                <button class="pswp__button pswp__button--fs" title="Toggle fullscreen"></button>
                <button class="pswp__button pswp__button--zoom" title="Zoom in/out"></button>
                <!-- Preloader demo http://codepen.io/dimsemenov/pen/yyBWoR -->
                <!-- element will get class pswp__preloader--active when preloader is running -->
                <div class="pswp__preloader">
                    <div class="pswp__preloader__icn">
                      <div class="pswp__preloader__cut">
                        <div class="pswp__preloader__donut"></div>
                      </div>
                    </div>
                </div>
            </div>
            <div class="pswp__share-modal pswp__share-modal--hidden pswp__single-tap">
                <div class="pswp__share-tooltip"></div> 
            </div>
            <button class="pswp__button pswp__button--arrow--left" title="Previous (arrow left)">
            </button>
            <button class="pswp__button pswp__button--arrow--right" title="Next (arrow right)">
            </button>
            <div class="pswp__caption">
                <div class="pswp__caption__center"></div>
            </div>
        </div>
    </div>
</div>






</html>
