
<!DOCTYPE html>
<html lang="en" class="loading">
<head>
    <meta charset="UTF-8" />
    <meta http-equiv="X-UA-Compatible" content="IE=edge,chrome=1" />
    <meta name="viewport" content="width=device-width, minimum-scale=1.0, maximum-scale=1.0, user-scalable=no">
    <title>生信技能树单细胞数据挖掘笔记(3) - Zhz Blog</title>
    <meta name="apple-mobile-web-app-capable" content="yes" />
    <meta name="apple-mobile-web-app-status-bar-style" content="black-translucent">
    <meta name="google" content="notranslate" />
    <meta name="keywords" content="zhoushizhao,"> 
    <meta name="description" content="生信技能树单细胞数据挖掘笔记(1)：数据读入
生信技能树单细胞数据挖掘笔记(2)：创建Seurat对象并进行质控、筛选高变基因并可视化
生信技能树单细胞数据挖掘笔记(3)：降维与聚类
生信技能树单细,"> 
    <meta name="author" content="John Doe"> 
    <link rel="alternative" href="atom.xml" title="Zhz Blog" type="application/atom+xml"> 
    <link rel="icon" href="/img/favicon.png"> 
    
    
    
    <meta name="twitter:card" content="summary"/>
    <meta name="twitter:title" content="生信技能树单细胞数据挖掘笔记(3) - Zhz Blog"/>
    <meta name="twitter:description" content="生信技能树单细胞数据挖掘笔记(1)：数据读入
生信技能树单细胞数据挖掘笔记(2)：创建Seurat对象并进行质控、筛选高变基因并可视化
生信技能树单细胞数据挖掘笔记(3)：降维与聚类
生信技能树单细,"/>
    
    
    
    
    <meta property="og:site_name" content="Zhz Blog"/>
    <meta property="og:type" content="object"/>
    <meta property="og:title" content="生信技能树单细胞数据挖掘笔记(3) - Zhz Blog"/>
    <meta property="og:description" content="生信技能树单细胞数据挖掘笔记(1)：数据读入
生信技能树单细胞数据挖掘笔记(2)：创建Seurat对象并进行质控、筛选高变基因并可视化
生信技能树单细胞数据挖掘笔记(3)：降维与聚类
生信技能树单细,"/>
    
<link rel="stylesheet" href="/css/diaspora.css">

<meta name="generator" content="Hexo 5.4.1"></head>

<body class="loading">
    <span id="config-title" style="display:none">Zhz Blog</span>
    <div id="loader"></div>
    <div id="single">
    <div id="top" style="display: block;">
    <div class="bar" style="width: 0;"></div>
    <a class="iconfont icon-home image-icon" href="javascript:;" data-url="http://example.com"></a>
    <div title="播放/暂停" class="iconfont icon-play"></div>
    <h3 class="subtitle">生信技能树单细胞数据挖掘笔记(3)</h3>
    <div class="social">
        <div>
            <div class="share">
                <a title="获取二维码" class="iconfont icon-scan" href="javascript:;"></a>
            </div>
            <div id="qr"></div>
        </div>
    </div>
    <div class="scrollbar"></div>
</div>

    <div class="section">
        <div class="article">
    <div class='main'>
        <h1 class="title">生信技能树单细胞数据挖掘笔记(3)</h1>
        <div class="stuff">
            <span>十月 29, 2020</span>
            
  <ul class="post-tags-list" itemprop="keywords"><li class="post-tags-list-item"><a class="post-tags-list-link" href="/tags/ScRNAseq/" rel="tag">ScRNAseq</a></li><li class="post-tags-list-item"><a class="post-tags-list-link" href="/tags/bioinformatics/" rel="tag">bioinformatics</a></li></ul>


        </div>
        <div class="content markdown">
            <p><a target="_blank" rel="noopener" href="https://www.zhouxiaozhao.cn/2020/10/24/ScRNAseq(7)/">生信技能树单细胞数据挖掘笔记(1)：数据读入</a></p>
<p><a target="_blank" rel="noopener" href="https://www.zhouxiaozhao.cn/2020/10/27/ScRNAseq(8)/">生信技能树单细胞数据挖掘笔记(2)：创建Seurat对象并进行质控、筛选高变基因并可视化</a></p>
<p><a target="_blank" rel="noopener" href="https://www.zhouxiaozhao.cn/2020/10/29/ScRNAseq(9)/">生信技能树单细胞数据挖掘笔记(3)：降维与聚类</a></p>
<p><a target="_blank" rel="noopener" href="https://www.zhouxiaozhao.cn/2020/10/31/ScRNAseq(10)/">生信技能树单细胞数据挖掘笔记(4)：其他分析（周期判断、double诊断、细胞类型注释）</a></p>
<p><a target="_blank" rel="noopener" href="https://www.zhouxiaozhao.cn/2020/11/03/ScRNAseq(11)/">生信技能树单细胞数据挖掘笔记(5)：轨迹分析</a></p>
<p><a target="_blank" rel="noopener" href="https://www.zhouxiaozhao.cn/2020/11/05/ScRNAseq(12)/">生信技能树单细胞数据挖掘笔记(6)：差异分析(1)</a></p>
<p><a target="_blank" rel="noopener" href="https://www.zhouxiaozhao.cn/2020/11/07/ScRNAseq(13)/">生信技能树单细胞数据挖掘笔记(7)：差异分析(2)</a></p>
<h2 id="降维，PCA分析"><a href="#降维，PCA分析" class="headerlink" title="降维，PCA分析"></a>降维，PCA分析</h2><figure class="highlight plaintext"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br><span class="line">7</span><br><span class="line">8</span><br><span class="line">9</span><br><span class="line">10</span><br><span class="line">11</span><br><span class="line">12</span><br><span class="line">13</span><br><span class="line">14</span><br><span class="line">15</span><br><span class="line">16</span><br><span class="line">17</span><br><span class="line">18</span><br><span class="line">19</span><br><span class="line">20</span><br><span class="line">21</span><br><span class="line">22</span><br><span class="line">23</span><br></pre></td><td class="code"><pre><span class="line">load(&quot;../2.3.Rdata&quot;)</span><br><span class="line">### 4、降维，PCA分析，可视化----</span><br><span class="line">#先进行归一化（正态分布）</span><br><span class="line">scRNA &lt;- ScaleData(scRNA, features = (rownames(scRNA)))</span><br><span class="line">#储存到&quot;scale.data&quot;的slot里</span><br><span class="line">GetAssayData(scRNA,slot=&quot;scale.data&quot;,assay=&quot;RNA&quot;)[1:8,1:4]</span><br><span class="line">#对比下原来的count矩阵</span><br><span class="line">GetAssayData(scRNA,slot=&quot;counts&quot;,assay=&quot;RNA&quot;)[1:8,1:4]</span><br><span class="line">#scRNA@assays$RNA@</span><br><span class="line">#PCA降维，利用之前挑选的hvg，可提高效率</span><br><span class="line">scRNA &lt;- RunPCA(scRNA, features = VariableFeatures(scRNA))</span><br><span class="line">#挑选第一，第二主成分对cell可视化</span><br><span class="line">DimPlot(scRNA, reduction = &quot;pca&quot;, group.by=&quot;Patient_ID&quot;)</span><br><span class="line">#发现与原文献中颠倒了</span><br><span class="line">#seed.use	:Set a random seed. By default, sets the seed to 42.</span><br><span class="line">#Setting NULL will not set a seed.</span><br><span class="line">scRNA &lt;- RunPCA(scRNA, features = VariableFeatures(scRNA),seed.use=3)</span><br><span class="line">#尝试了seed.use的不同取值发现图形只有四种变化（四个拐角），其中以seed.use=3为代表的一类与原文文献一致</span><br><span class="line">DimPlot(scRNA, reduction = &quot;pca&quot;, group.by=&quot;Patient_ID&quot;)</span><br><span class="line">#与文献一致了。个人觉得颠倒与否如果只是随机种子的差别的话，对后续分析应该没影响</span><br><span class="line">p2_1 &lt;- DimPlot(scRNA, reduction = &quot;pca&quot;, group.by=&quot;Patient_ID&quot;)+</span><br><span class="line">  labs(tag = &quot;D&quot;)</span><br><span class="line">p2_1</span><br></pre></td></tr></table></figure>

<p><img src="/img/posts/2020.10.29/image-20201109102111288.png" alt="image-20201109102111288"></p>
<center>
    原图
</center>

<p><img src="/img/posts/2020.10.29/image-20201109102200544.png" alt="image-20201109102200544"></p>
<center>
    seed.use=3后的PCA图
</center>

<figure class="highlight plaintext"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br><span class="line">7</span><br><span class="line">8</span><br><span class="line">9</span><br><span class="line">10</span><br></pre></td><td class="code"><pre><span class="line">#挑选主成分，RunPCA默认保留了前50个</span><br><span class="line">scRNA &lt;- JackStraw(scRNA,reduction = &quot;pca&quot;, dims=20)</span><br><span class="line">scRNA &lt;- ScoreJackStraw(scRNA,dims = 1:20)</span><br><span class="line"></span><br><span class="line">p2_2 &lt;- JackStrawPlot(scRNA,dims = 1:20, reduction = &quot;pca&quot;) +</span><br><span class="line">  theme(legend.position=&quot;bottom&quot;) +</span><br><span class="line">  labs(tag = &quot;E&quot;)</span><br><span class="line">p2_2</span><br><span class="line">p2_3 &lt;- ElbowPlot(scRNA, ndims=20, reduction=&quot;pca&quot;)</span><br><span class="line">p2_2 | p2_3</span><br></pre></td></tr></table></figure>

<p><img src="/img/posts/2020.10.29/image-20201109104437291.png" alt="image-20201109104437291"></p>
<center>
    E图
</center>

<p><img src="/img/posts/2020.10.29/image-20201109104530915.png" alt="image-20201109104530915"></p>
<center>
    D图与E图
</center>

<h2 id="聚类、筛选maker基因并可视化"><a href="#聚类、筛选maker基因并可视化" class="headerlink" title="聚类、筛选maker基因并可视化"></a>聚类、筛选maker基因并可视化</h2><h3 id="聚类"><a href="#聚类" class="headerlink" title="聚类"></a>聚类</h3><figure class="highlight plaintext"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br><span class="line">7</span><br><span class="line">8</span><br><span class="line">9</span><br><span class="line">10</span><br><span class="line">11</span><br><span class="line">12</span><br></pre></td><td class="code"><pre><span class="line">pc.num=1:20</span><br><span class="line">#基于PCA数据</span><br><span class="line">scRNA &lt;- FindNeighbors(scRNA, dims = pc.num)</span><br><span class="line"># dims参数，需要指定哪些pc轴用于分析；这里利用上面的分析，选择20</span><br><span class="line">scRNA &lt;- FindClusters(scRNA, resolution = 0.5)</span><br><span class="line">table(scRNA@meta.data$seurat_clusters)</span><br><span class="line"></span><br><span class="line">scRNA = RunTSNE(scRNA, dims = pc.num)</span><br><span class="line">DimPlot(scRNA, reduction = &quot;tsne&quot;,label=T)</span><br><span class="line">p3_1 &lt;- DimPlot(scRNA, reduction = &quot;tsne&quot;,label=T) +</span><br><span class="line">  labs(tag = &quot;E&quot;)</span><br><span class="line">p3_1</span><br></pre></td></tr></table></figure>

<p><img src="/img/posts/2020.10.29/image-20201109152047996.png" alt="image-20201109152047996"></p>
<center>
    F图
</center>

<h3 id="筛选maker基因"><a href="#筛选maker基因" class="headerlink" title="筛选maker基因"></a>筛选maker基因</h3><figure class="highlight plaintext"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br><span class="line">7</span><br><span class="line">8</span><br><span class="line">9</span><br><span class="line">10</span><br><span class="line">11</span><br><span class="line">12</span><br><span class="line">13</span><br><span class="line">14</span><br><span class="line">15</span><br><span class="line">16</span><br><span class="line">17</span><br><span class="line">18</span><br><span class="line">19</span><br><span class="line">20</span><br><span class="line">21</span><br><span class="line">22</span><br><span class="line">23</span><br><span class="line">24</span><br><span class="line">25</span><br><span class="line">26</span><br><span class="line">27</span><br><span class="line">28</span><br><span class="line">29</span><br><span class="line">30</span><br></pre></td><td class="code"><pre><span class="line">#5.2 marker gene</span><br><span class="line">#进行差异分析，一般使用标准化数据</span><br><span class="line">scRNA &lt;- NormalizeData(scRNA, normalization.method = &quot;LogNormalize&quot;)</span><br><span class="line">#结果储存在&quot;data&quot;slot里</span><br><span class="line">GetAssayData(scRNA,slot=&quot;data&quot;,assay=&quot;RNA&quot;)[1:8,1:4]</span><br><span class="line"></span><br><span class="line">#if test.use is &quot;negbinom&quot;, &quot;poisson&quot;, or &quot;DESeq2&quot;, slot will be set to &quot;counts</span><br><span class="line">diff.wilcox = FindAllMarkers(scRNA)##默认使用wilcox方法挑选差异基因，大概4-5min</span><br><span class="line">load(&quot;../diff.wilcox.Rdata&quot;)</span><br><span class="line">head(diff.wilcox)</span><br><span class="line">dim(diff.wilcox)</span><br><span class="line">library(tidyverse)</span><br><span class="line">all.markers = diff.wilcox %&gt;% select(gene, everything()) %&gt;%</span><br><span class="line">  subset(p_val&lt;0.05 &amp; abs(diff.wilcox$avg_logFC) &gt; 0.5)</span><br><span class="line">#An adjusted P value &lt; 0.05and | log 2 [fold change (FC)] | &gt; 0.5</span><br><span class="line">#were considered the 2 cutoff criteria for identifying marker genes.</span><br><span class="line">dim(all.markers)</span><br><span class="line">summary(all.markers)</span><br><span class="line">save(all.markers,file = &quot;../markergene.Rdata&quot;)</span><br><span class="line"></span><br><span class="line">top10 = all.markers %&gt;% group_by(cluster) %&gt;% top_n(n = 10, wt = avg_logFC)</span><br><span class="line">top10</span><br><span class="line">top10 = CaseMatch(search = as.vector(top10$gene), match = rownames(scRNA))</span><br><span class="line">top10</span><br><span class="line">length(top10)</span><br><span class="line">length(unique(sort(top10)))</span><br><span class="line"></span><br><span class="line">p3_2 &lt;- DoHeatmap(scRNA, features = top10, group.by = &quot;seurat_clusters&quot;)</span><br><span class="line">p3_2</span><br><span class="line">p3_1 | p3_2 #下图</span><br></pre></td></tr></table></figure>

<p><img src="/img/posts/2020.10.29/image-20201109153125281.png" alt="image-20201109153125281"></p>
<center>
    G图
</center>

<p><img src="/img/posts/2020.10.29/image-20201109153150765.png" alt="image-20201109153150765"></p>
<center>
    图片合并
</center>

<p>目前，图一的图片已经全部完成，接下来将进行拼图</p>
<h2 id="拼图"><a href="#拼图" class="headerlink" title="拼图"></a>拼图</h2><figure class="highlight plaintext"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br><span class="line">7</span><br></pre></td><td class="code"><pre><span class="line">### 6、拼图，比较----</span><br><span class="line">p &lt;- (p1_1 | p1_2 | p1_3 ) /</span><br><span class="line">  ((p2_1| p2_2 | p2_3) /</span><br><span class="line">     (p3_1 | p3_2))</span><br><span class="line">ggsave(&quot;../.my_try.pdf&quot;, plot = p, width = 15, height = 18)</span><br><span class="line"></span><br><span class="line">save(scRNA,file = &quot;scRNA.Rdata&quot;)</span><br></pre></td></tr></table></figure>

<p><img src="/img/posts/2020.10.29/image-20201109153745153.png" alt="image-20201109153745153"></p>

            <!--[if lt IE 9]><script>document.createElement('audio');</script><![endif]-->
            <audio id="audio" loop="1" preload="auto" controls="controls" data-autoplay="false">
                <source type="audio/mpeg" src="">
            </audio>
            
                <ul id="audio-list" style="display:none">
                    
                        
                            <li title="0" data-url="http://link.hhtjim.com/163/425570952.mp3"></li>
                        
                    
                        
                            <li title="1" data-url="http://link.hhtjim.com/163/425570952.mp3"></li>
                        
                    
                </ul>
            
        </div>
        
    <div id="gitalk-container" class="comment link"
		data-enable="false"
        data-ae="false"
        data-ci=""
        data-cs=""
        data-r=""
        data-o=""
        data-a=""
        data-d="false"
    >查看评论</div>


    </div>
    
</div>


    </div>
</div>
</body>


<script src="//lib.baomitu.com/jquery/1.8.3/jquery.min.js"></script>
<script src="/js/plugin.js"></script>
<script src="/js/typed.js"></script>
<script src="/js/diaspora.js"></script>


<link rel="stylesheet" href="/photoswipe/photoswipe.css">
<link rel="stylesheet" href="/photoswipe/default-skin/default-skin.css">


<script src="/photoswipe/photoswipe.min.js"></script>
<script src="/photoswipe/photoswipe-ui-default.min.js"></script>


<!-- Root element of PhotoSwipe. Must have class pswp. -->
<div class="pswp" tabindex="-1" role="dialog" aria-hidden="true">
    <!-- Background of PhotoSwipe. 
         It's a separate element as animating opacity is faster than rgba(). -->
    <div class="pswp__bg"></div>
    <!-- Slides wrapper with overflow:hidden. -->
    <div class="pswp__scroll-wrap">
        <!-- Container that holds slides. 
            PhotoSwipe keeps only 3 of them in the DOM to save memory.
            Don't modify these 3 pswp__item elements, data is added later on. -->
        <div class="pswp__container">
            <div class="pswp__item"></div>
            <div class="pswp__item"></div>
            <div class="pswp__item"></div>
        </div>
        <!-- Default (PhotoSwipeUI_Default) interface on top of sliding area. Can be changed. -->
        <div class="pswp__ui pswp__ui--hidden">
            <div class="pswp__top-bar">
                <!--  Controls are self-explanatory. Order can be changed. -->
                <div class="pswp__counter"></div>
                <button class="pswp__button pswp__button--close" title="Close (Esc)"></button>
                <button class="pswp__button pswp__button--share" title="Share"></button>
                <button class="pswp__button pswp__button--fs" title="Toggle fullscreen"></button>
                <button class="pswp__button pswp__button--zoom" title="Zoom in/out"></button>
                <!-- Preloader demo http://codepen.io/dimsemenov/pen/yyBWoR -->
                <!-- element will get class pswp__preloader--active when preloader is running -->
                <div class="pswp__preloader">
                    <div class="pswp__preloader__icn">
                      <div class="pswp__preloader__cut">
                        <div class="pswp__preloader__donut"></div>
                      </div>
                    </div>
                </div>
            </div>
            <div class="pswp__share-modal pswp__share-modal--hidden pswp__single-tap">
                <div class="pswp__share-tooltip"></div> 
            </div>
            <button class="pswp__button pswp__button--arrow--left" title="Previous (arrow left)">
            </button>
            <button class="pswp__button pswp__button--arrow--right" title="Next (arrow right)">
            </button>
            <div class="pswp__caption">
                <div class="pswp__caption__center"></div>
            </div>
        </div>
    </div>
</div>






</html>
