
<!DOCTYPE html>
<html lang="en" class="loading">
<head>
    <meta charset="UTF-8" />
    <meta http-equiv="X-UA-Compatible" content="IE=edge,chrome=1" />
    <meta name="viewport" content="width=device-width, minimum-scale=1.0, maximum-scale=1.0, user-scalable=no">
    <title>根据转录组数据鉴定共生物种 - Zhz Blog</title>
    <meta name="apple-mobile-web-app-capable" content="yes" />
    <meta name="apple-mobile-web-app-status-bar-style" content="black-translucent">
    <meta name="google" content="notranslate" />
    <meta name="keywords" content="zhoushizhao,"> 
    <meta name="description" content="最近做的项目是关于地衣的转录组和基因组，拿到的结果是真菌部位的基因组以及不同时期和部位的转录组，地下部位的藻类部由于太复杂而且杂质过多不好分离就没有测基因组，目前想知道地下部位的藻类到底是什么？根据,"> 
    <meta name="author" content="John Doe"> 
    <link rel="alternative" href="atom.xml" title="Zhz Blog" type="application/atom+xml"> 
    <link rel="icon" href="/img/favicon.png"> 
    
    
    
    <meta name="twitter:card" content="summary"/>
    <meta name="twitter:title" content="根据转录组数据鉴定共生物种 - Zhz Blog"/>
    <meta name="twitter:description" content="最近做的项目是关于地衣的转录组和基因组，拿到的结果是真菌部位的基因组以及不同时期和部位的转录组，地下部位的藻类部由于太复杂而且杂质过多不好分离就没有测基因组，目前想知道地下部位的藻类到底是什么？根据,"/>
    
    
    
    
    <meta property="og:site_name" content="Zhz Blog"/>
    <meta property="og:type" content="object"/>
    <meta property="og:title" content="根据转录组数据鉴定共生物种 - Zhz Blog"/>
    <meta property="og:description" content="最近做的项目是关于地衣的转录组和基因组，拿到的结果是真菌部位的基因组以及不同时期和部位的转录组，地下部位的藻类部由于太复杂而且杂质过多不好分离就没有测基因组，目前想知道地下部位的藻类到底是什么？根据,"/>
    
<link rel="stylesheet" href="/css/diaspora.css">

<meta name="generator" content="Hexo 5.4.1"></head>

<body class="loading">
    <span id="config-title" style="display:none">Zhz Blog</span>
    <div id="loader"></div>
    <div id="single">
    <div id="top" style="display: block;">
    <div class="bar" style="width: 0;"></div>
    <a class="iconfont icon-home image-icon" href="javascript:;" data-url="http://example.com"></a>
    <div title="播放/暂停" class="iconfont icon-play"></div>
    <h3 class="subtitle">根据转录组数据鉴定共生物种</h3>
    <div class="social">
        <div>
            <div class="share">
                <a title="获取二维码" class="iconfont icon-scan" href="javascript:;"></a>
            </div>
            <div id="qr"></div>
        </div>
    </div>
    <div class="scrollbar"></div>
</div>

    <div class="section">
        <div class="article">
    <div class='main'>
        <h1 class="title">根据转录组数据鉴定共生物种</h1>
        <div class="stuff">
            <span>八月 18, 2020</span>
            
  <ul class="post-tags-list" itemprop="keywords"><li class="post-tags-list-item"><a class="post-tags-list-link" href="/tags/bioinformatics/" rel="tag">bioinformatics</a></li><li class="post-tags-list-item"><a class="post-tags-list-link" href="/tags/transcriptome/" rel="tag">transcriptome</a></li></ul>


        </div>
        <div class="content markdown">
            <p>最近做的项目是关于地衣的转录组和基因组，拿到的结果是真菌部位的基因组以及不同时期和部位的转录组，地下部位的藻类部由于太复杂而且杂质过多不好分离就没有测基因组，目前想知道地下部位的藻类到底是什么？根据之前的观察，推测是<em>Coccomyxa subellipsoidea</em>，想要具体确定的话还需要进一步证明。</p>
<p><img src="/img/posts/2020.8.18/image-20200826100329224.png"></p>
<p>可以看到该地衣的地上部位是一种担子菌，地下部位不仅有藻类还有其他的物种。</p>
<h2 id="转录组比对到真菌参考基因组"><a href="#转录组比对到真菌参考基因组" class="headerlink" title="转录组比对到真菌参考基因组"></a>转录组比对到真菌参考基因组</h2><p>目前有真菌的测序基因组，第一步就是想把转录组数据比对到真菌的参考基因组上</p>
<p>比对工具我选择STAR，使用地上部位（真菌部_Lh3-1）和地下部位（藻类部 _Lh1-1）的转录组比对到基因组上</p>
<p><strong>建立索引</strong></p>
<figure class="highlight plaintext"><table><tr><td class="gutter"><pre><span class="line">1</span><br></pre></td><td class="code"><pre><span class="line">S$ATR --runMode genomeGenerate --genomeDir /datadisk02/mpoly/hun_db/ --genomeFastaFiles /datadisk02/mpoly/hudsoiana.fa</span><br></pre></td></tr></table></figure>

<p><strong>比对</strong></p>
<figure class="highlight plaintext"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br></pre></td><td class="code"><pre><span class="line">$STAR --runThreadN 20 --genomeDir /datadisk02/mpoly/hun_db/ --readFilesCommand zcat --readFilesIn _Lh2-1_S110_L007_R1_001.fastq.gz _Lh2-1_S110_L007_R2_001.fastq.gz --outSAMtype BAM SortedByCoordinate --outBAMsortingThreadN 5 --outFileNamePrefix S2-1</span><br><span class="line"></span><br><span class="line">$STAR --runThreadN 20 --genomeDir /datadisk02/mpoly/hun_db/ --readFilesCommand zcat --readFilesIn _Lh3-1_S108_L007_R1_001.fastq.gz _Lh3-1_S108_L007_R2_001.fastq.gz --outSAMtype BAM SortedByCoordinate --outBAMsortingThreadN 5 --outFileNamePrefix S1-1</span><br></pre></td></tr></table></figure>

<p>比对结束后我最关心的是比对率，按照预期，地下部位的比对率比地上部位低。</p>
<figure class="highlight plaintext"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br><span class="line">7</span><br><span class="line">8</span><br><span class="line">9</span><br><span class="line">10</span><br><span class="line">11</span><br><span class="line">12</span><br><span class="line">13</span><br><span class="line">14</span><br><span class="line">15</span><br><span class="line">16</span><br><span class="line">17</span><br><span class="line">18</span><br><span class="line">19</span><br><span class="line">20</span><br><span class="line">21</span><br><span class="line">22</span><br><span class="line">23</span><br><span class="line">24</span><br></pre></td><td class="code"><pre><span class="line">$less S1-1Log.final.out</span><br><span class="line">                                 Started job on |       Aug 20 05:48:00</span><br><span class="line">                             Started mapping on |       Aug 20 05:48:01</span><br><span class="line">                                    Finished on |       Aug 20 06:42:35</span><br><span class="line">       Mapping speed, Million of reads per hour |       26.61</span><br><span class="line"></span><br><span class="line">                          Number of input reads |       24198466</span><br><span class="line">                      Average input read length |       300</span><br><span class="line">                                    UNIQUE READS:</span><br><span class="line">                   Uniquely mapped reads number |       15397710</span><br><span class="line">                        Uniquely mapped reads % |       60.89%</span><br><span class="line"></span><br><span class="line">$less S3-1Log.final.out</span><br><span class="line">                                 Started job on |       Aug 20 08:38:29</span><br><span class="line">                             Started mapping on |       Aug 20 08:38:30</span><br><span class="line">                                    Finished on |       Aug 20 08:49:15</span><br><span class="line">       Mapping speed, Million of reads per hour |       134.06</span><br><span class="line"></span><br><span class="line">                          Number of input reads |       24018884</span><br><span class="line">                      Average input read length |       300</span><br><span class="line">                                    UNIQUE READS:</span><br><span class="line">                   Uniquely mapped reads number |       19151198</span><br><span class="line">                        Uniquely mapped reads % |       79.73%</span><br><span class="line"></span><br></pre></td></tr></table></figure>

<p>可以看到结果符合预期，地下部位的reads比对率为64%，地上部位为80%。</p>
<h2 id="地下部位转录组分析"><a href="#地下部位转录组分析" class="headerlink" title="地下部位转录组分析"></a>地下部位转录组分析</h2><p>接下来用地下部位的转录组分析确定共生藻，首先的想法是利用转录组数据比对到初步确定的共生藻<em>Coccomyxa subellipsoidea</em>上，使用<em>Coccomyxa subellipsoidea</em>的基因组建库进行比对</p>
<figure class="highlight plaintext"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br></pre></td><td class="code"><pre><span class="line">S$ATR --runMode genomeGenerate --genomeDir /datadisk02/mpoly/cocco/ --genomeFastaFiles /datadisk02/mpoly/Coccomyxa.fa</span><br><span class="line"></span><br><span class="line">$STAR --runThreadN 20 --genomeDir /datadisk02/mpoly/cocco/ --readFilesCommand zcat --readFilesIn _Lh1-1_S110_L007_R1_001.fastq.gz _Lh1-1_S110_L007_R2_001.fastq.gz --outSAMtype BAM SortedByCoordinate --outBAMsortingThreadN 5 --outFileNamePrefix S1-1C</span><br></pre></td></tr></table></figure>

<p>结果有点出乎意料，只有零点几的比对率，</p>
<figure class="highlight plaintext"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br><span class="line">7</span><br><span class="line">8</span><br><span class="line">9</span><br><span class="line">10</span><br><span class="line">11</span><br><span class="line">12</span><br></pre></td><td class="code"><pre><span class="line">$less S1-1CLog.final.out</span><br><span class="line">                                 Started job on |       Aug 19 14:25:11</span><br><span class="line">                             Started mapping on |       Aug 19 14:25:16</span><br><span class="line">                                    Finished on |       Aug 19 20:51:41</span><br><span class="line">       Mapping speed, Million of reads per hour |       3.96</span><br><span class="line"></span><br><span class="line">                          Number of input reads |       25517519</span><br><span class="line">                      Average input read length |       300</span><br><span class="line">                                    UNIQUE READS:</span><br><span class="line">                   Uniquely mapped reads number |       92457</span><br><span class="line">                        Uniquely mapped reads % |       0.36%</span><br><span class="line"></span><br></pre></td></tr></table></figure>

<p>由于地下部位比对到真菌部也有64%的比对率，所以我重新跑了一遍STAR，利用unmapped reads进行比对。</p>
<figure class="highlight plaintext"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br><span class="line">7</span><br><span class="line">8</span><br><span class="line">9</span><br><span class="line">10</span><br><span class="line">11</span><br><span class="line">12</span><br><span class="line">13</span><br><span class="line">14</span><br><span class="line">15</span><br><span class="line">16</span><br><span class="line">17</span><br><span class="line">18</span><br><span class="line">19</span><br><span class="line">20</span><br></pre></td><td class="code"><pre><span class="line">$STAR --runThreadN 32 --genomeDir hun_db/ --readFilesIn _Lh1-1_S110_L007_R1_001.fastq.gz _Lh1-1_S110_L007_R2_001.fastq.gz --readFilesCommand zcat --outSAMtype SAM --outFileNamePrefix S1-1 --outReadsUnmapped Fastx</span><br><span class="line"></span><br><span class="line">$awk -F &quot; &quot; &#x27;&#123;print $1&#125;&#x27; S1-1Unmapped.out.mate1| awk &#x27;&#123;if (NR%4==1||NR%4=3) print&#125;&#x27; &gt; S1-1.left.fa</span><br><span class="line">$sed -i &quot;s/@/&gt;/g&quot; S1-1.left.fa</span><br><span class="line">$awk -F &quot; &quot; &#x27;&#123;print $1&#125;&#x27; S1-1Unmapped.out.mate2| awk &#x27;&#123;if (NR%4==1||NR%4=3) print&#125;&#x27; &gt; S1-1.right.fa</span><br><span class="line">$sed -i &quot;s/@/&gt;/g&quot; S1-1.right.fa</span><br><span class="line">$STAR --runThreadN 20 --genomeDir cocco/ --readFilesIn S1-1.left.fa S1-1.right.fa --outSAMtype SAM --outFileNamePrefix S1-1UC --outReadsUnmapped Fastx</span><br><span class="line"></span><br><span class="line">$less S1-1UCLog.final.out</span><br><span class="line"></span><br><span class="line">                                 Started job on |       Aug 26 10:59:26</span><br><span class="line">                             Started mapping on |       Aug 26 10:59:39</span><br><span class="line">                                    Finished on |       Aug 26 15:32:14</span><br><span class="line">       Mapping speed, Million of reads per hour |       1.58</span><br><span class="line"></span><br><span class="line">                          Number of input reads |       7180320</span><br><span class="line">                      Average input read length |       300</span><br><span class="line">                                    UNIQUE READS:</span><br><span class="line">                   Uniquely mapped reads number |       96710</span><br><span class="line">                        Uniquely mapped reads % |       1.35%</span><br></pre></td></tr></table></figure>

<p>比对率也只有可怜的1.35%，一开始我怀疑是不是藻类物种鉴定错了，后来我用了ncbi中所有藻类的基因组，比对率都不到1%。</p>
<p>后来去网上搜索比对率低的原因，在<a target="_blank" rel="noopener" href="https://github.com/alexdobin/STAR/issues/169">github</a>上发现相应解答,加入–outFilterScoreMinOverLread 0.3和–outFilterMatchNminOverLread 0.3 参数</p>
<figure class="highlight plaintext"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br><span class="line">7</span><br><span class="line">8</span><br><span class="line">9</span><br><span class="line">10</span><br><span class="line">11</span><br><span class="line">12</span><br><span class="line">13</span><br><span class="line">14</span><br><span class="line">15</span><br><span class="line">16</span><br><span class="line">17</span><br><span class="line">18</span><br><span class="line">19</span><br><span class="line">20</span><br><span class="line">21</span><br><span class="line">22</span><br><span class="line">23</span><br><span class="line">24</span><br><span class="line">25</span><br><span class="line">26</span><br><span class="line">27</span><br><span class="line">28</span><br></pre></td><td class="code"><pre><span class="line">$STAR --runThreadN 20 --genomeDir hun_db/ --readFilesCommand zcat --readFilesIn _Lh1-1_S110_L007_R1_001.fastq.gz _Lh1-1_S110_L007_R2_001.fastq.gz --outFileNamePrefix S1-1fli --outReadsUnmapped Fastx --outSAMtype SAM --outFilterScoreMinOverLread 0.3 --outFilterMatchNminOverLread 0.3</span><br><span class="line"></span><br><span class="line">$less S1-1fliLog.finall.out</span><br><span class="line">                                 Started job on |       Aug 27 09:52:31</span><br><span class="line">                             Started mapping on |       Aug 27 09:52:37</span><br><span class="line">                                    Finished on |       Aug 27 13:02:11</span><br><span class="line">       Mapping speed, Million of reads per hour |       8.08</span><br><span class="line"></span><br><span class="line">                          Number of input reads |       25517519</span><br><span class="line">                      Average input read length |       300</span><br><span class="line">                                    UNIQUE READS:</span><br><span class="line">                   Uniquely mapped reads number |       16371292</span><br><span class="line">                        Uniquely mapped reads % |       64.16%</span><br><span class="line"></span><br><span class="line">$STAR --runThreadN 20 --genomeDir cocco/ --readFilesCommand zcat --readFilesIn _Lh1-1_S110_L007_R1_001.fastq.gz _Lh1-1_S110_L007_R2_001.fastq.gz --outFileNamePrefix S1-1Cfli --outReadsUnmapped Fastx --outSAMtype SAM --outFilterScoreMinOverLread 0.3 --outFilterMatchNminOverLread 0.3</span><br><span class="line"></span><br><span class="line">$less S1-1CfliLog.final.out</span><br><span class="line">                                 Started job on |       Aug 27 09:52:51</span><br><span class="line">                             Started mapping on |       Aug 27 09:52:53</span><br><span class="line">                                    Finished on |       Aug 28 04:02:41</span><br><span class="line">       Mapping speed, Million of reads per hour |       1.40</span><br><span class="line"></span><br><span class="line">                          Number of input reads |       25517519</span><br><span class="line">                      Average input read length |       300</span><br><span class="line">                                    UNIQUE READS:</span><br><span class="line">                   Uniquely mapped reads number |       1475579</span><br><span class="line">                        Uniquely mapped reads % |       5.78%</span><br><span class="line"></span><br></pre></td></tr></table></figure>

<p>对率有5.78%，提高了一点。</p>
<p>接下来，我使用真菌的unmapped reads进行比对，看看没有比到真菌上的reads有多少是藻类的。</p>
<figure class="highlight plaintext"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br><span class="line">7</span><br><span class="line">8</span><br><span class="line">9</span><br><span class="line">10</span><br><span class="line">11</span><br><span class="line">12</span><br><span class="line">13</span><br><span class="line">14</span><br><span class="line">15</span><br><span class="line">16</span><br><span class="line">17</span><br><span class="line">18</span><br><span class="line">19</span><br></pre></td><td class="code"><pre><span class="line">$awk -F &quot; &quot; &#x27;&#123;print $1&#125;&#x27; S1-1fliUnmapped.out.mate1| awk &#x27;&#123;if (NR%4==1||NR%4=3) print&#125;&#x27; &gt; S1-1fli.left.fa</span><br><span class="line">$sed -i &quot;s/@/&gt;/g&quot; S1-1.left.fa</span><br><span class="line">$awk -F &quot; &quot; &#x27;&#123;print $1&#125;&#x27; S1-1fliUnmapped.out.mate2| awk &#x27;&#123;if (NR%4==1||NR%4=3) print&#125;&#x27; &gt; S1-1fli.right.fa</span><br><span class="line">$sed -i &quot;s/@/&gt;/g&quot; S1-1.right.fa</span><br><span class="line">$STAR --runThreadN 20 --genomeDir cocco/ --readFilesIn S1-1fli.left.fa S1-1fli.right.fa --outSAMtype SAM --outFileNamePrefix S1-1Ufli --outReadsUnmapped Fastx --outFilterScoreMinOverLread 0.3 --outFilterMatchNminOverLread 0.3</span><br><span class="line"></span><br><span class="line">$less S1-1UfliLog.final.out</span><br><span class="line">                                 Started job on |       Aug 27 14:28:39</span><br><span class="line">                             Started mapping on |       Aug 27 14:28:44</span><br><span class="line">                                    Finished on |       Aug 27 21:38:00</span><br><span class="line">       Mapping speed, Million of reads per hour |       0.86</span><br><span class="line"></span><br><span class="line">                          Number of input reads |       6123885</span><br><span class="line">                      Average input read length |       300</span><br><span class="line">                                    UNIQUE READS:</span><br><span class="line">                   Uniquely mapped reads number |       1441904</span><br><span class="line">                        Uniquely mapped reads % |       23.55%</span><br><span class="line"></span><br><span class="line"></span><br></pre></td></tr></table></figure>

<p>也就是说，在剩余的36%未比对的reads中有23%属于Coccomyxa，总比对率有70%左右，地上部位按照相同的步骤进行比对有83%的比对率，这个现象也正常，因为地下部位的组织特别复杂，在取样时肯定包含了很多杂质，比对率肯定比地上部位低。</p>
<h2 id="总结"><a href="#总结" class="headerlink" title="总结"></a>总结</h2><p>目前已经可以确定地下部位的藻类就是Coccomyxa属的，是不是<em>Coccomyxa subellipsoidea</em>目前还不确定，接下来的工作是确定地下部位未比对上reads的所属。</p>

            <!--[if lt IE 9]><script>document.createElement('audio');</script><![endif]-->
            <audio id="audio" loop="1" preload="auto" controls="controls" data-autoplay="false">
                <source type="audio/mpeg" src="">
            </audio>
            
                <ul id="audio-list" style="display:none">
                    
                        
                            <li title="0" data-url="http://link.hhtjim.com/163/425570952.mp3"></li>
                        
                    
                        
                            <li title="1" data-url="http://link.hhtjim.com/163/425570952.mp3"></li>
                        
                    
                </ul>
            
        </div>
        
    <div id="gitalk-container" class="comment link"
		data-enable="false"
        data-ae="false"
        data-ci=""
        data-cs=""
        data-r=""
        data-o=""
        data-a=""
        data-d="false"
    >查看评论</div>


    </div>
    
</div>


    </div>
</div>
</body>


<script src="//lib.baomitu.com/jquery/1.8.3/jquery.min.js"></script>
<script src="/js/plugin.js"></script>
<script src="/js/typed.js"></script>
<script src="/js/diaspora.js"></script>


<link rel="stylesheet" href="/photoswipe/photoswipe.css">
<link rel="stylesheet" href="/photoswipe/default-skin/default-skin.css">


<script src="/photoswipe/photoswipe.min.js"></script>
<script src="/photoswipe/photoswipe-ui-default.min.js"></script>


<!-- Root element of PhotoSwipe. Must have class pswp. -->
<div class="pswp" tabindex="-1" role="dialog" aria-hidden="true">
    <!-- Background of PhotoSwipe. 
         It's a separate element as animating opacity is faster than rgba(). -->
    <div class="pswp__bg"></div>
    <!-- Slides wrapper with overflow:hidden. -->
    <div class="pswp__scroll-wrap">
        <!-- Container that holds slides. 
            PhotoSwipe keeps only 3 of them in the DOM to save memory.
            Don't modify these 3 pswp__item elements, data is added later on. -->
        <div class="pswp__container">
            <div class="pswp__item"></div>
            <div class="pswp__item"></div>
            <div class="pswp__item"></div>
        </div>
        <!-- Default (PhotoSwipeUI_Default) interface on top of sliding area. Can be changed. -->
        <div class="pswp__ui pswp__ui--hidden">
            <div class="pswp__top-bar">
                <!--  Controls are self-explanatory. Order can be changed. -->
                <div class="pswp__counter"></div>
                <button class="pswp__button pswp__button--close" title="Close (Esc)"></button>
                <button class="pswp__button pswp__button--share" title="Share"></button>
                <button class="pswp__button pswp__button--fs" title="Toggle fullscreen"></button>
                <button class="pswp__button pswp__button--zoom" title="Zoom in/out"></button>
                <!-- Preloader demo http://codepen.io/dimsemenov/pen/yyBWoR -->
                <!-- element will get class pswp__preloader--active when preloader is running -->
                <div class="pswp__preloader">
                    <div class="pswp__preloader__icn">
                      <div class="pswp__preloader__cut">
                        <div class="pswp__preloader__donut"></div>
                      </div>
                    </div>
                </div>
            </div>
            <div class="pswp__share-modal pswp__share-modal--hidden pswp__single-tap">
                <div class="pswp__share-tooltip"></div> 
            </div>
            <button class="pswp__button pswp__button--arrow--left" title="Previous (arrow left)">
            </button>
            <button class="pswp__button pswp__button--arrow--right" title="Next (arrow right)">
            </button>
            <div class="pswp__caption">
                <div class="pswp__caption__center"></div>
            </div>
        </div>
    </div>
</div>






</html>
