
<!DOCTYPE html>
<html lang="en" class="loading">
<head>
    <meta charset="UTF-8" />
    <meta http-equiv="X-UA-Compatible" content="IE=edge,chrome=1" />
    <meta name="viewport" content="width=device-width, minimum-scale=1.0, maximum-scale=1.0, user-scalable=no">
    <title>基因结构注释（2）：同源注释 - Zhz Blog</title>
    <meta name="apple-mobile-web-app-capable" content="yes" />
    <meta name="apple-mobile-web-app-status-bar-style" content="black-translucent">
    <meta name="google" content="notranslate" />
    <meta name="keywords" content="zhoushizhao,"> 
    <meta name="description" content="同源预测(homology prediction)利用近缘物种已知基因进行序列比对，找到同源序列。然后在同源序列的基础上，根据基因信号如剪切信号、基因起始和终止密码子对基因结构进行预测.
在同源预测,"> 
    <meta name="author" content="John Doe"> 
    <link rel="alternative" href="atom.xml" title="Zhz Blog" type="application/atom+xml"> 
    <link rel="icon" href="/img/favicon.png"> 
    
    
    
    <meta name="twitter:card" content="summary"/>
    <meta name="twitter:title" content="基因结构注释（2）：同源注释 - Zhz Blog"/>
    <meta name="twitter:description" content="同源预测(homology prediction)利用近缘物种已知基因进行序列比对，找到同源序列。然后在同源序列的基础上，根据基因信号如剪切信号、基因起始和终止密码子对基因结构进行预测.
在同源预测,"/>
    
    
    
    
    <meta property="og:site_name" content="Zhz Blog"/>
    <meta property="og:type" content="object"/>
    <meta property="og:title" content="基因结构注释（2）：同源注释 - Zhz Blog"/>
    <meta property="og:description" content="同源预测(homology prediction)利用近缘物种已知基因进行序列比对，找到同源序列。然后在同源序列的基础上，根据基因信号如剪切信号、基因起始和终止密码子对基因结构进行预测.
在同源预测,"/>
    
<link rel="stylesheet" href="/css/diaspora.css">

<meta name="generator" content="Hexo 5.4.1"></head>

<body class="loading">
    <span id="config-title" style="display:none">Zhz Blog</span>
    <div id="loader"></div>
    <div id="single">
    <div id="top" style="display: block;">
    <div class="bar" style="width: 0;"></div>
    <a class="iconfont icon-home image-icon" href="javascript:;" data-url="http://example.com"></a>
    <div title="播放/暂停" class="iconfont icon-play"></div>
    <h3 class="subtitle">基因结构注释（2）：同源注释</h3>
    <div class="social">
        <div>
            <div class="share">
                <a title="获取二维码" class="iconfont icon-scan" href="javascript:;"></a>
            </div>
            <div id="qr"></div>
        </div>
    </div>
    <div class="scrollbar"></div>
</div>

    <div class="section">
        <div class="article">
    <div class='main'>
        <h1 class="title">基因结构注释（2）：同源注释</h1>
        <div class="stuff">
            <span>十一月 21, 2020</span>
            
  <ul class="post-tags-list" itemprop="keywords"><li class="post-tags-list-item"><a class="post-tags-list-link" href="/tags/bioinformatics/" rel="tag">bioinformatics</a></li><li class="post-tags-list-item"><a class="post-tags-list-link" href="/tags/gene-annotion/" rel="tag">gene_annotion</a></li></ul>


        </div>
        <div class="content markdown">
            <p>同源预测(homology prediction)利用近缘物种已知基因进行序列比对，找到同源序列。然后在同源序列的基础上，根据基因信号如剪切信号、基因起始和终止密码子对基因结构进行预测.</p>
<p>在同源预测上，目前看到的大部分基因组文章都是基于TBLASTN + GeneWise,但是目前GeneWise已经不在维护，在本次推文中将使用GenomeThreader进行同源注释。</p>
<p>参考链接：<a target="_blank" rel="noopener" href="https://www.jianshu.com/p/931e9821c45a">如何对基因组注释</a></p>
<h2 id="确定同源物种蛋白序列"><a href="#确定同源物种蛋白序列" class="headerlink" title="确定同源物种蛋白序列"></a>确定同源物种蛋白序列</h2><p>首先选择同源物种，在本次分析中，我使用Saccharomyces_cerevisiae、Laccaria_bicolor、Amanita_thiersii、Pleurotus_pulmonarius、Pterula_gracilis进行同源注释</p>
<figure class="highlight plaintext"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br></pre></td><td class="code"><pre><span class="line">cat Saccharomyces_cerevisiae.fa\       </span><br><span class="line">    Laccaria_bicolor.fa \</span><br><span class="line">    Amanita_thiersii \</span><br><span class="line">    Pleurotus_pulmonarius \</span><br><span class="line">    Pterula_gracilis &gt;all.pep.fa</span><br></pre></td></tr></table></figure>

<p>然后使用TBLASTN确定匹配到基因组的蛋白序列</p>
<figure class="highlight plaintext"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br><span class="line">7</span><br><span class="line">8</span><br><span class="line">9</span><br><span class="line">10</span><br><span class="line">11</span><br><span class="line">12</span><br><span class="line">13</span><br><span class="line">14</span><br></pre></td><td class="code"><pre><span class="line">makeblastdb -in pudorinus.fa -parse_seqids -dbtype nucl -out index/pu&amp;</span><br><span class="line">## tblastn比对</span><br><span class="line">nohup tblastn -query all.pep.fa -out pu.blast -db index/pu -outfmt 6 -evalue 1e-10 -num_threads 8 -qcov_hsp_perc 50.0 -num_alignments 5 &amp;</span><br><span class="line">## 提取匹配到基因组的蛋白序列</span><br><span class="line">awk &#x27;&#123;print $1&#125;&#x27; pu.blast &gt;pu.list</span><br><span class="line">sort pu.list| unique &gt;pi.ho.list</span><br><span class="line">seqkit seq  all.pep.fa   -w   0   &gt;  all.fa</span><br><span class="line">vi sh.sh</span><br><span class="line">   cat list.ru | while read line</span><br><span class="line">   do</span><br><span class="line">   grep &quot;$line&quot; -A 1 all.fa &gt;$line.1</span><br><span class="line">   done</span><br><span class="line">   cat *.1 &gt; pudorinus.homo.fa</span><br><span class="line">   rm -rf *.1</span><br></pre></td></tr></table></figure>

<p>使用gth进行同源注释</p>
<figure class="highlight plaintext"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br></pre></td><td class="code"><pre><span class="line">gth -genomic pudorinus.fa -protein pudorinus.homo.fa -intermediate -gff3out &gt; pudorinus.gff</span><br><span class="line"></span><br></pre></td></tr></table></figure>

<p><img src="/img/posts/2020.11.21/image-20201209151007890.png" alt="image-20201209151007890"></p>
<p>删除一些无用的信息可以看到基本已经注释完成，但是exon之类的没有ID，在后续识别会有问题</p>
<figure class="highlight plaintext"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br></pre></td><td class="code"><pre><span class="line">grep -v &quot;^#&quot; pudorinus.gth.gff | grep -v &quot;prime_cis_splice_site&quot; | awk -F &quot;;&quot; &#x27;&#123;print$1&#125;&#x27;&gt;pudorinus.homo.gff</span><br><span class="line">less pudorinus.homo.gff</span><br></pre></td></tr></table></figure>

<p><img src="/img/posts/2020.11.21/image-20201209151032555.png" alt="image-20201209151032555"></p>
<p>使用脚本处理下(谢谢课题组师姐帮忙写的脚本(#^.^#))</p>
<figure class="highlight plaintext"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br><span class="line">7</span><br><span class="line">8</span><br><span class="line">9</span><br><span class="line">10</span><br><span class="line">11</span><br><span class="line">12</span><br><span class="line">13</span><br><span class="line">14</span><br><span class="line">15</span><br><span class="line">16</span><br><span class="line">17</span><br><span class="line">18</span><br><span class="line">19</span><br><span class="line">20</span><br><span class="line">21</span><br><span class="line">22</span><br><span class="line">23</span><br><span class="line">24</span><br><span class="line">25</span><br><span class="line">26</span><br><span class="line">27</span><br><span class="line">28</span><br><span class="line">29</span><br><span class="line">30</span><br><span class="line">31</span><br><span class="line">32</span><br><span class="line">33</span><br><span class="line">34</span><br><span class="line">35</span><br><span class="line">36</span><br><span class="line">37</span><br><span class="line">38</span><br><span class="line">39</span><br><span class="line">40</span><br><span class="line">41</span><br><span class="line">42</span><br><span class="line">43</span><br><span class="line">44</span><br></pre></td><td class="code"><pre><span class="line">#!/lustre/home/guohan_lab/local/python-3.6/bin/python3</span><br><span class="line">#liyumei</span><br><span class="line">#./change-name_lym.py proteinprediction.gff proteinprediction.gff proteinprediction_r.gff</span><br><span class="line"></span><br><span class="line">import sys</span><br><span class="line">import re</span><br><span class="line">fout = open(sys.argv[3],&#x27;w&#x27;)</span><br><span class="line"></span><br><span class="line">ref_dict=&#123;&#125;</span><br><span class="line">with open(sys.argv[1]) as gff:</span><br><span class="line">        for line in gff:</span><br><span class="line">                line_s = line.strip().split(&#x27;\t&#x27;)</span><br><span class="line">                if &#x27;gene&#x27; == line_s[2]:</span><br><span class="line">                        ref_g = re.split(&#x27;=&#x27;,line_s[8])</span><br><span class="line">                        ref_gene = ref_g[1]</span><br><span class="line">                        ref_dict[ref_gene]=[]</span><br><span class="line">                else:</span><br><span class="line">                        pos = line_s[3]</span><br><span class="line">                        ref_dict[ref_gene].append(pos)</span><br><span class="line"></span><br><span class="line">with open(sys.argv[2]) as gff_r:</span><br><span class="line">        for eachline in gff_r:</span><br><span class="line">                i = eachline.strip().split(&#x27;\t&#x27;)</span><br><span class="line">                info = re.split(&#x27;=&#x27;,i[8])</span><br><span class="line">                name = info[1]</span><br><span class="line">                ref_set = []</span><br><span class="line">                for n in range(0,8):</span><br><span class="line">                        ref_set.append(i[n])</span><br><span class="line">                ref_list = &#x27;\t&#x27;.join(ref_set)</span><br><span class="line">                ref_set1 = [&#x27;CDS&#x27; if x == &#x27;exon&#x27; else x for x in ref_set]</span><br><span class="line">                ref_list1 = &#x27;\t&#x27;.join(ref_set1)</span><br><span class="line">                if &#x27;gene&#x27; == i[2]:</span><br><span class="line">                        fout.write(eachline)</span><br><span class="line">                elif &#x27;exon&#x27; == i[2]:</span><br><span class="line">                        if name in ref_dict:</span><br><span class="line">                                vs = ref_dict[name]</span><br><span class="line">                                len_vs = len(vs)</span><br><span class="line">                                for a in range(0,len_vs):</span><br><span class="line">                                        if vs[a] == i[3]:</span><br><span class="line">                                                b = vs.index(vs[a])</span><br><span class="line">                                                fout.write(&#x27;%s\tID=%s.t%d;Parent=%s\n&#x27;%(ref_list,name,b+1,name))</span><br><span class="line">                                                fout.write(&#x27;%s\tID=%s.c%d;Parent=%s\n&#x27;%(ref_list1,name,b+1,name))</span><br><span class="line">fout.close()</span><br><span class="line"></span><br></pre></td></tr></table></figure>

<figure class="highlight plaintext"><table><tr><td class="gutter"><pre><span class="line">1</span><br></pre></td><td class="code"><pre><span class="line">python3 ../annotion/change-gff_lym.py pudorinus.homo.gff pudorinus.homo.gff prediction.pudorinus.gff</span><br></pre></td></tr></table></figure>

<p><img src="/img/posts/2020.11.21/image-20201209151108664.png" alt="image-20201209151108664"></p>
<p>至此同源注释已经结束</p>

            <!--[if lt IE 9]><script>document.createElement('audio');</script><![endif]-->
            <audio id="audio" loop="1" preload="auto" controls="controls" data-autoplay="false">
                <source type="audio/mpeg" src="">
            </audio>
            
                <ul id="audio-list" style="display:none">
                    
                        
                            <li title="0" data-url="http://link.hhtjim.com/163/425570952.mp3"></li>
                        
                    
                        
                            <li title="1" data-url="http://link.hhtjim.com/163/425570952.mp3"></li>
                        
                    
                </ul>
            
        </div>
        
    <div id="gitalk-container" class="comment link"
		data-enable="false"
        data-ae="false"
        data-ci=""
        data-cs=""
        data-r=""
        data-o=""
        data-a=""
        data-d="false"
    >查看评论</div>


    </div>
    
</div>


    </div>
</div>
</body>


<script src="//lib.baomitu.com/jquery/1.8.3/jquery.min.js"></script>
<script src="/js/plugin.js"></script>
<script src="/js/typed.js"></script>
<script src="/js/diaspora.js"></script>


<link rel="stylesheet" href="/photoswipe/photoswipe.css">
<link rel="stylesheet" href="/photoswipe/default-skin/default-skin.css">


<script src="/photoswipe/photoswipe.min.js"></script>
<script src="/photoswipe/photoswipe-ui-default.min.js"></script>


<!-- Root element of PhotoSwipe. Must have class pswp. -->
<div class="pswp" tabindex="-1" role="dialog" aria-hidden="true">
    <!-- Background of PhotoSwipe. 
         It's a separate element as animating opacity is faster than rgba(). -->
    <div class="pswp__bg"></div>
    <!-- Slides wrapper with overflow:hidden. -->
    <div class="pswp__scroll-wrap">
        <!-- Container that holds slides. 
            PhotoSwipe keeps only 3 of them in the DOM to save memory.
            Don't modify these 3 pswp__item elements, data is added later on. -->
        <div class="pswp__container">
            <div class="pswp__item"></div>
            <div class="pswp__item"></div>
            <div class="pswp__item"></div>
        </div>
        <!-- Default (PhotoSwipeUI_Default) interface on top of sliding area. Can be changed. -->
        <div class="pswp__ui pswp__ui--hidden">
            <div class="pswp__top-bar">
                <!--  Controls are self-explanatory. Order can be changed. -->
                <div class="pswp__counter"></div>
                <button class="pswp__button pswp__button--close" title="Close (Esc)"></button>
                <button class="pswp__button pswp__button--share" title="Share"></button>
                <button class="pswp__button pswp__button--fs" title="Toggle fullscreen"></button>
                <button class="pswp__button pswp__button--zoom" title="Zoom in/out"></button>
                <!-- Preloader demo http://codepen.io/dimsemenov/pen/yyBWoR -->
                <!-- element will get class pswp__preloader--active when preloader is running -->
                <div class="pswp__preloader">
                    <div class="pswp__preloader__icn">
                      <div class="pswp__preloader__cut">
                        <div class="pswp__preloader__donut"></div>
                      </div>
                    </div>
                </div>
            </div>
            <div class="pswp__share-modal pswp__share-modal--hidden pswp__single-tap">
                <div class="pswp__share-tooltip"></div> 
            </div>
            <button class="pswp__button pswp__button--arrow--left" title="Previous (arrow left)">
            </button>
            <button class="pswp__button pswp__button--arrow--right" title="Next (arrow right)">
            </button>
            <div class="pswp__caption">
                <div class="pswp__caption__center"></div>
            </div>
        </div>
    </div>
</div>






</html>
