
<!DOCTYPE html>
<html lang="en" class="loading">
<head>
    <meta charset="UTF-8" />
    <meta http-equiv="X-UA-Compatible" content="IE=edge,chrome=1" />
    <meta name="viewport" content="width=device-width, minimum-scale=1.0, maximum-scale=1.0, user-scalable=no">
    <title>使用本地nt数据库对reads和Trinity结果进行blast - Zhz Blog</title>
    <meta name="apple-mobile-web-app-capable" content="yes" />
    <meta name="apple-mobile-web-app-status-bar-style" content="black-translucent">
    <meta name="google" content="notranslate" />
    <meta name="keywords" content="zhoushizhao,"> 
    <meta name="description" content="上个笔记中，进行了共生物种的确定，由于地下部位的转录组还有一部分reads没有比对上，可能是样品污染问题，也可能含有其他的物种，所以，想使用Trinity和为比对上的reads去比对到nt数据库查看,"> 
    <meta name="author" content="John Doe"> 
    <link rel="alternative" href="atom.xml" title="Zhz Blog" type="application/atom+xml"> 
    <link rel="icon" href="/img/favicon.png"> 
    
    
    
    <meta name="twitter:card" content="summary"/>
    <meta name="twitter:title" content="使用本地nt数据库对reads和Trinity结果进行blast - Zhz Blog"/>
    <meta name="twitter:description" content="上个笔记中，进行了共生物种的确定，由于地下部位的转录组还有一部分reads没有比对上，可能是样品污染问题，也可能含有其他的物种，所以，想使用Trinity和为比对上的reads去比对到nt数据库查看,"/>
    
    
    
    
    <meta property="og:site_name" content="Zhz Blog"/>
    <meta property="og:type" content="object"/>
    <meta property="og:title" content="使用本地nt数据库对reads和Trinity结果进行blast - Zhz Blog"/>
    <meta property="og:description" content="上个笔记中，进行了共生物种的确定，由于地下部位的转录组还有一部分reads没有比对上，可能是样品污染问题，也可能含有其他的物种，所以，想使用Trinity和为比对上的reads去比对到nt数据库查看,"/>
    
<link rel="stylesheet" href="/css/diaspora.css">

<meta name="generator" content="Hexo 5.4.1"></head>

<body class="loading">
    <span id="config-title" style="display:none">Zhz Blog</span>
    <div id="loader"></div>
    <div id="single">
    <div id="top" style="display: block;">
    <div class="bar" style="width: 0;"></div>
    <a class="iconfont icon-home image-icon" href="javascript:;" data-url="http://example.com"></a>
    <div title="播放/暂停" class="iconfont icon-play"></div>
    <h3 class="subtitle">使用本地nt数据库对reads和Trinity结果进行blast</h3>
    <div class="social">
        <div>
            <div class="share">
                <a title="获取二维码" class="iconfont icon-scan" href="javascript:;"></a>
            </div>
            <div id="qr"></div>
        </div>
    </div>
    <div class="scrollbar"></div>
</div>

    <div class="section">
        <div class="article">
    <div class='main'>
        <h1 class="title">使用本地nt数据库对reads和Trinity结果进行blast</h1>
        <div class="stuff">
            <span>八月 20, 2020</span>
            
  <ul class="post-tags-list" itemprop="keywords"><li class="post-tags-list-item"><a class="post-tags-list-link" href="/tags/bioinformatics/" rel="tag">bioinformatics</a></li><li class="post-tags-list-item"><a class="post-tags-list-link" href="/tags/transcriptome/" rel="tag">transcriptome</a></li></ul>


        </div>
        <div class="content markdown">
            <p>上个笔记中，进行了共生物种的确定，由于地下部位的转录组还有一部分reads没有比对上，可能是样品污染问题，也可能含有其他的物种，所以，想使用Trinity和为比对上的reads去比对到nt数据库查看结果</p>
<h2 id="nt数据库下载和构建"><a href="#nt数据库下载和构建" class="headerlink" title="nt数据库下载和构建"></a>nt数据库下载和构建</h2><figure class="highlight plaintext"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br></pre></td><td class="code"><pre><span class="line">wget https://ftp.ncbi.nih.gov/blast/db/FASTA/nt.gz</span><br><span class="line">wget https://ftp.ncbi.nih.gov/blast/db/FASTA/nr.gz.md5</span><br><span class="line">md5sum nt.gz</span><br><span class="line">gunzip nt.gz</span><br><span class="line">nohup makeblastdb -in nt -parse_seqids -hash_index -dbtype nucl -logfile nt_logfile &amp;</span><br></pre></td></tr></table></figure>

<p>这样，nt数据库就构建好了，后续的话利用这个数据库去确定物种</p>
<p>下载数据库和构建时间有点长，需要耐心等待</p>
<h2 id="使用未比对的reads进行blast"><a href="#使用未比对的reads进行blast" class="headerlink" title="使用未比对的reads进行blast"></a>使用未比对的reads进行blast</h2><p>在运行STAR时，加入–outReadsUnmapped Fastx参数会将未比对的reads输出到文件，双端测序会生成mate1和mate2两个文件，利用该reads去blast</p>
<figure class="highlight plaintext"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br></pre></td><td class="code"><pre><span class="line">$cat S1-1Ufli.left.fa S1-1Ufli.right.fa&gt;S1-1.reads.fa</span><br><span class="line">$cat S3-1fli.left.fa S3-1fli.right.fa&gt;S3-1.reads.fa</span><br><span class="line"></span><br><span class="line">$nohup blastn -query S1-1.reads.fa -out S1-1.reads.blast -db /datadisk02/data/nt -outfmt 6 -evalue 1e-10 -num_threads 8 -qcov_hsp_perc 50.0 -num_alignments 5 2&gt; blastn1-1.log &amp;</span><br><span class="line"></span><br><span class="line">$nohup blastn -query S3-1.reads.fa -out S3-1.reads.blast -db /datadisk02/data/nt -outfmt 6 -evalue 1e-10 -num_threads 8 -qcov_hsp_perc 50.0 -num_alignments 5 2&gt; blastn3-1.log &amp;</span><br></pre></td></tr></table></figure>

<h2 id="使用Trinity组装未比对的reads后进行blast"><a href="#使用Trinity组装未比对的reads后进行blast" class="headerlink" title="使用Trinity组装未比对的reads后进行blast"></a>使用Trinity组装未比对的reads后进行blast</h2><p>在把未比对的reads进行blast之后，我又试着把未比对的reads用Trinity进行组装，并进行blast</p>
<figure class="highlight plaintext"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br><span class="line">7</span><br></pre></td><td class="code"><pre><span class="line">$nohup Trinity --seqType fa --max_memory 50G --left S1-1Ufli.left.fa  --right S1-1Ufli.right.fa --CPU 16 --output S1-1Ufli_trinity &amp;</span><br><span class="line"></span><br><span class="line">$nohup Trinity --seqType fa --max_memory 50G --left S3-1fli.left.fa  --right S3-1fli.right.fa --CPU 16 --output S3-1fli_trinity &amp;</span><br><span class="line"></span><br><span class="line">$nohup blastn -query S1-1Ufli_trinity/Trinity.fasta -out S1-1.trinity.blast -db /datadisk02/data/nt -outfmt 6 -evalue 1e-10 -num_threads 8 -qcov_hsp_perc 50.0 -num_alignments 5 2&gt; blastn1-1t.log &amp;</span><br><span class="line"></span><br><span class="line">$nohup blastn -query S3-1fli_trinity/Trinity.fasta -out S3-1.trinity.blast -db /datadisk02/data/nt -outfmt 6 -evalue 1e-10 -num_threads 8 -qcov_hsp_perc 50.0 -num_alignments 5 2&gt; blastn3-1t.log &amp;</span><br></pre></td></tr></table></figure>

<h2 id="查看blast结果"><a href="#查看blast结果" class="headerlink" title="查看blast结果"></a>查看blast结果</h2><p>对于blast结果，主要是对比对到的基因进行汇总，去找哪个物种比对到的基因最多，涉及课题原因比对到的物种我就不在这里展示了。</p>

            <!--[if lt IE 9]><script>document.createElement('audio');</script><![endif]-->
            <audio id="audio" loop="1" preload="auto" controls="controls" data-autoplay="false">
                <source type="audio/mpeg" src="">
            </audio>
            
                <ul id="audio-list" style="display:none">
                    
                        
                            <li title="0" data-url="http://link.hhtjim.com/163/425570952.mp3"></li>
                        
                    
                        
                            <li title="1" data-url="http://link.hhtjim.com/163/425570952.mp3"></li>
                        
                    
                </ul>
            
        </div>
        
    <div id="gitalk-container" class="comment link"
		data-enable="false"
        data-ae="false"
        data-ci=""
        data-cs=""
        data-r=""
        data-o=""
        data-a=""
        data-d="false"
    >查看评论</div>


    </div>
    
</div>


    </div>
</div>
</body>


<script src="//lib.baomitu.com/jquery/1.8.3/jquery.min.js"></script>
<script src="/js/plugin.js"></script>
<script src="/js/typed.js"></script>
<script src="/js/diaspora.js"></script>


<link rel="stylesheet" href="/photoswipe/photoswipe.css">
<link rel="stylesheet" href="/photoswipe/default-skin/default-skin.css">


<script src="/photoswipe/photoswipe.min.js"></script>
<script src="/photoswipe/photoswipe-ui-default.min.js"></script>


<!-- Root element of PhotoSwipe. Must have class pswp. -->
<div class="pswp" tabindex="-1" role="dialog" aria-hidden="true">
    <!-- Background of PhotoSwipe. 
         It's a separate element as animating opacity is faster than rgba(). -->
    <div class="pswp__bg"></div>
    <!-- Slides wrapper with overflow:hidden. -->
    <div class="pswp__scroll-wrap">
        <!-- Container that holds slides. 
            PhotoSwipe keeps only 3 of them in the DOM to save memory.
            Don't modify these 3 pswp__item elements, data is added later on. -->
        <div class="pswp__container">
            <div class="pswp__item"></div>
            <div class="pswp__item"></div>
            <div class="pswp__item"></div>
        </div>
        <!-- Default (PhotoSwipeUI_Default) interface on top of sliding area. Can be changed. -->
        <div class="pswp__ui pswp__ui--hidden">
            <div class="pswp__top-bar">
                <!--  Controls are self-explanatory. Order can be changed. -->
                <div class="pswp__counter"></div>
                <button class="pswp__button pswp__button--close" title="Close (Esc)"></button>
                <button class="pswp__button pswp__button--share" title="Share"></button>
                <button class="pswp__button pswp__button--fs" title="Toggle fullscreen"></button>
                <button class="pswp__button pswp__button--zoom" title="Zoom in/out"></button>
                <!-- Preloader demo http://codepen.io/dimsemenov/pen/yyBWoR -->
                <!-- element will get class pswp__preloader--active when preloader is running -->
                <div class="pswp__preloader">
                    <div class="pswp__preloader__icn">
                      <div class="pswp__preloader__cut">
                        <div class="pswp__preloader__donut"></div>
                      </div>
                    </div>
                </div>
            </div>
            <div class="pswp__share-modal pswp__share-modal--hidden pswp__single-tap">
                <div class="pswp__share-tooltip"></div> 
            </div>
            <button class="pswp__button pswp__button--arrow--left" title="Previous (arrow left)">
            </button>
            <button class="pswp__button pswp__button--arrow--right" title="Next (arrow right)">
            </button>
            <div class="pswp__caption">
                <div class="pswp__caption__center"></div>
            </div>
        </div>
    </div>
</div>






</html>
